src/Rings/MPolyQuo.jl

```diff
@@ -126,23 +151,34 @@ Quotient of Multivariate Polynomial Ring in x, y, z over Rational Field by ideal
 ```
 """
 function base_ring(a::MPolyQuoIdeal)
-  return a.base_ring
+  return a.qRing
 end
 
-# TODO rename oscar_assure to oscar_assure!
 function oscar_assure(a::MPolyQuoIdeal)
-  isdefined(a, :I) && return
+  if isdefined(a.gens.gens, :O)
+      return a.gens.gens.O
+  end
   r = base_ring(base_ring(a))
-  a.I = ideal(r, r.(gens(a.SI)))
+  a.gens.O = [r(g) for g = gens(a.gens.S)]
 end
```

Comment:
If it is there, we return `a.gens.gens.O`. If not we set `a.gens.O` and return `a.gens.O`. Is this on purpose?

Reply:
Fixed accordingly.